lib/LogPriceLib.sol

```diff
@@ -52,11 +53,28 @@ library LogPriceLib {
   }
 
   function outboundFromInboundUp(int logPrice, uint inboundAmt) internal pure returns (uint) {
-    (uint sig, uint exp) = LogPriceConversionLib.nonNormalizedPriceFromLogPrice(-logPrice);
-    uint prod = sig*inboundAmt;
-    return (prod>>exp) + (prod%(1<<exp)==0 ? 0 : 1);
+    unchecked {
+      (uint sig, uint exp) = LogPriceConversionLib.nonNormalizedPriceFromLogPrice(-logPrice);
+      return divExpUp(sig*inboundAmt,exp);
+    }
   }
+}
 
+// Return a/2**e rounded up
+function divExpUp(uint a, uint e) pure returns (uint) {
+  unchecked {
+    uint carry;
+    /* 
+    Would like to write
 
+      carry = (e<<1 == 0 || prod % (e<<1) != 0) ? 1 : 0
 
+    but solidity reverts on a%0 even inside unchecked blocks
+    */
+    assembly("memory-safe") {
+      let den := shl(e,1)
+      carry := gt(or(mod(a,den),iszero(den)),0)
```

Comment:
When can `iszero(den) == iszero(shl(e,1))` be true? When `e > 255`?

Reply:
If so, shouldn't there only be a carry if `a > 0`?

Reply:
pushed a fix :)